pkg/volume/plugins.go

```diff
@@ -119,6 +119,39 @@ type VolumePluginMgr struct {
 	plugins map[string]VolumePlugin
 }
 
+// InternalVolume is an internal representation of a volume.  All API volume types translate to InternalVolume.
+type InternalVolume struct {
```

Comment:
The pointers complicated the CanSupport methods in the plugins, actually, because they'd have to check first for nil VS/PSV and then the nil src.